connection.js

```diff
@@ -1369,6 +1302,35 @@ function pluralFamilyStringToSingularString(plural) { /* {{{ */
     return pluralToConvert;
 } /* }}} */
 
+function rankedUnrankedGamesString(argNameString) { /* {{{ */
+    // for the "bans" family :
+    if argNameString.includes("bans") {
```

Comment:
A pattern like this where a function contains one big if/else that both return is an indication it should have been 2 functions. Is it possible here to make a separate bans function and make the distinction between bans and others elsewhere?

You will also want to avoid to make the bans check everywhere else, but in places where you already know it'll always be bans or never be bans, you could call the right split function. If you end up having to do that bans check in multiple places, you could make 3 functions, one that does the ban check, and then calls the appropriate one of the other 2.

A quick scan trough the code shows we have `bannedFamilyReject`, that's probably the only one that needs to call the ban format string right?

Reply:
ok, then since the "bans" family is only one case, we'll just keep the rankedunranked specific code in the banned function
not worth creating one function that we'll use only in 1 case

Reply:
It's fine like this, but i like to itinerate that functions that do one small thing are good, even if they are only used once. For more background information, see this list: https://herbertograca.com/2016/09/03/clean-code-3-functions-by-robert-c-martin-uncle-bob/